### PR TITLE
fix annotations for Python >= 3.8

### PR DESCRIPTION
**Pull request by pmeier**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ def write_version_file():
     pytorch_dep += "==" + os.getenv("PYTORCH_VERSION")
 
 requirements = [
+    "typing_extensions",
```
Comment (pmeier) [inline setup.py:61]: PyTorch core already hard-depends on it: https://github.com/pytorch/pytorch/blob/be2dc8f2940d3c95941516a811be8c504910d1ea/setup.py#L408-L411

**Review by pmeier — COMMENTED** [review]
These changes are necessary when linting against `numpy>=1.22`, which is the latest version for Python >= 3.8. This was not caught by our CI, because we run 3.7 there.

File: torchvision/transforms/functional_pil.py [hunk 2]
```diff
@@ -130,7 +131,7 @@ def pad(
     img: Image.Image,
     padding: Union[int, List[int], Tuple[int, ...]],
     fill: Optional[Union[float, List[float], Tuple[float, ...]]] = 0,
-    padding_mode: str = "constant",
+    padding_mode: Literal["constant", "edge", "reflect", "symmetric"] = "constant",
```
Comment (datumbox) [inline torchvision/transforms/functional_pil.py:134]: That's a new one for me, `Literally`.

**Review by datumbox — APPROVED** [review]
LGTM!

File: torchvision/datasets/stl10.py [hunk 3]
```diff
@@ -65,10 +65,12 @@ def __init__(
         self.labels: Optional[np.ndarray]
         if self.split == "train":
             self.data, self.labels = self.__loadfile(self.train_list[0][0], self.train_list[1][0])
+            self.labels = cast(np.ndarray, self.labels)
```
Comment (NicolasHug) [inline torchvision/datasets/stl10.py:68]: We already have `self.labels: Optional[np.ndarray]` above, should we remove one of these?

Comment (pmeier) [reply torchvision/datasets/stl10.py:68]: I missed that. Let me have a look.
